src/services/TeamAssignmentService.ts

```diff
@@ -0,0 +1,49 @@
+import { PlayerInActiveMatch } from "../repositories/ActiveMatchRepository/types";
+import { PlayerInQueue } from "../repositories/QueueRepository/types";
+import { Team } from "../types/common";
+
+export async function createRandomTeams(
+  ballchasers: ReadonlyArray<PlayerInQueue>
+): Promise<Array<PlayerInActiveMatch>> {
+  const sortedBallChaser = ballchasers
+    .map((player) => {
```

Comment:
Instead of doing a map like this you can just do a `.slice()` to get a copy of the array.

Reply:
Good idea.  Fixed in next commit.